tools/remap/convert_remap_actions_to_10x

```diff
@@ -0,0 +1,172 @@
+        input_dir = '.' if input_dir == '' else input_dir
+        input_base = os.path.basename(input)
+        # Find the largest <num> in input_dir/input_base.action_to_add.bak.<num>.
+        max_num = 0
+        for file in os.listdir(input_dir):
+            if file.startswith(f'{input_base}.action_to_add.bak.'):
+                try:
+                    num = int(file.split('.')[-1])
+                except ValueError:
+                    # The suffix is not a number. Ignore this file.
+                    continue
+                max_num = max(max_num, num)
+        copy_path = os.path.join(input_dir, f'{input_base}.action_to_add.bak.{max_num + 1}')
+    shutil.copyfile(input, copy_path)
+    return copy_path
+
+
+def prepare_files(input: str, output: Optional[str], no_backup: bool) -> Tuple[str, str]:
+    '''Prepare the input and output files.
+
+    :param input: The input file path.
+    :param output: The output file path.
+    :param no_backup: Do not create a backup of the input file when <input> is modified.
+
+    :return: A tuple containing the input and output filenames, in that order.
+    The input file will contain the content from which to copy, the output file
+    will reference a file whose content should be overridden. The caller should
+    delete the input filename if @a no_backup is true. Otherwise the input
+    filename will be the path to the backup file.
+    '''
+    if not os.path.exists(input):
+        raise FileNotFoundError(f'Input file does not exist: {input}')
+
+    if input == output:
+        input_path = copy_input_file(input, no_backup)
+        output_path = input
+    elif output is None:
+        input_path = copy_input_file(input, no_backup)
+        output_path = input
+    else:
+        input_path = input
+        output_path = output
+    return input_path, output_path
+
+
+def parse_args() -> argparse.Namespace:
+    '''Parse command line arguments.
+
+    :return: The parsed arguments.
+    '''
+    parser = argparse.ArgumentParser(description=help)
+    parser.add_argument('input', help='The input remap.config file to modify.')
+    parser.add_argument(
+        '-o',
+        '--output',
+        help='The output remap.config file. By default, output is written to <input> and a backup file is generated.')
+    parser.add_argument(
+        '-n', '--no-backup', action='store_true', help='Do not create a backup of the input file when <input> is modified.')
+    return parser.parse_args()
+
+
+def main() -> int:
+    '''Convert the input remap.config file.
+
+    :return: The process exit code.
+    '''
+
+    args = parse_args()
+    input_path, output_path = prepare_files(args.input, args.output, args.no_backup)
+
+    with open(input_path, 'r') as fd_in, open(output_path, 'w') as fd_out:
+        for input_line in fd_in:
+            output_line = convert_line(input_line)
+            fd_out.write(output_line)
+    if args.no_backup:
+        os.remove(input_path)
```

Comment:
The backup won't get removed if an exception occurs after creating the file but before getting here, and that would be annoying. 😠